pkg/backup/client.go

```diff
@@ -833,45 +818,47 @@ func SendBackup(
 }
 
 // ChecksumMatches tests whether the "local" checksum matches the checksum from TiKV.
-func ChecksumMatches(backupMeta *kvproto.BackupMeta, local []Checksum) (bool, error) {
+func ChecksumMatches(backupMeta *kvproto.BackupMeta, local []Checksum) error {
 	if len(local) != len(backupMeta.Schemas) {
-		return false, nil
+		return berrors.ErrBackupChecksumMismatch.GenWithStack(
+			"checksum mismatch, checksum len %d, schema len %d", len(local), len(backupMeta.Schemas))
 	}
 
 	for i, schema := range backupMeta.Schemas {
 		localChecksum := local[i]
 		dbInfo := &model.DBInfo{}
 		err := json.Unmarshal(schema.Db, dbInfo)
 		if err != nil {
-			log.Error("failed in fast checksum, and cannot parse db info.")
-			return false, err
+			return berrors.ErrBackupChecksumMismatch.GenWithStackByArgs(
+				"failed in checksum, and cannot parse db info")
 		}
 		tblInfo := &model.TableInfo{}
 		err = json.Unmarshal(schema.Table, tblInfo)
 		if err != nil {
-			log.Error("failed in fast checksum, and cannot parse table info.")
-			return false, err
+			return berrors.ErrBackupChecksumMismatch.GenWithStackByArgs(
+				"failed in checksum, and cannot parse table info")
 		}
 		if localChecksum.Crc64Xor != schema.Crc64Xor ||
 			localChecksum.TotalBytes != schema.TotalBytes ||
 			localChecksum.TotalKvs != schema.TotalKvs {
-			log.Error("failed in fast checksum",
+			log.Error("checksum mismatch",
 				zap.Stringer("db", dbInfo.Name),
 				zap.Stringer("table", tblInfo.Name),
 				zap.Uint64("origin tidb crc64", schema.Crc64Xor),
 				zap.Uint64("calculated crc64", localChecksum.Crc64Xor),
 				zap.Uint64("origin tidb total kvs", schema.TotalKvs),
 				zap.Uint64("calculated total kvs", localChecksum.TotalKvs),
 				zap.Uint64("origin tidb total bytes", schema.TotalBytes),
-				zap.Uint64("calculated total bytes", localChecksum.TotalBytes),
-			)
-			return false, nil
+				zap.Uint64("calculated total bytes", localChecksum.TotalBytes))
+			// TODO enhance error
+			return berrors.ErrBackupChecksumMismatch.GenWithStackByArgs(
+				"failed in checksum, and cannot parse table info")
 		}
-		log.Info("fast checksum success",
```

Comment:
We should leave the 'fast' here or change the `br_full_ddl` test script. Otherwise the latter would fail...

Reply:
I have replaced all "fast checksum" with "checksum".